### ISSUE TITLE
wp-graphql-1.2.5: Uncaught Error: Class 'WPGraphQL\Telemetry\Tracker' not found

### ISSUE THREAD

**Issue by rburgst** [wp-graphql/wp-graphql#1753]
After upgrading from 1.1.3 to 1.2.5 I am getting the following error:

```
[02-Mar-2021 07:51:36 UTC] PHP Fatal error:  Uncaught Error: Class 'WPGraphQL\Telemetry\Tracker' not found in /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php:234
Stack trace:
#0 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php(76): WPGraphQL->actions()
#1 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/wp-graphql.php(54): WPGraphQL::instance()
#2 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/wp-graphql.php(57): graphql_init()
#3 /var/www/html/wp-admin/includes/plugin.php(670): include_once('/var/www/html/w...')
#4 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/Plugin_Command.php(323): activate_plugin('wp-graphql-1.2....', '', NULL)
#5 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/WP_CLI/CommandWithUpgrade.php(245): Plugin_Command->activate(Array)
#6 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/Plugin_Command.php(785): WP_CLI\CommandWithUpgrade->install(Array, Array)
#7 [internal function]: Plugin_Command->install(Array, Array)
#8 phar:///usr/local/bin/wp/vendor/wp- in /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php on line 234
Fatal error: Uncaught Error: Class 'WPGraphQL\Telemetry\Tracker' not found in /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php:234
Stack trace:
#0 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php(76): WPGraphQL->actions()
#1 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/wp-graphql.php(54): WPGraphQL::instance()
#2 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/wp-graphql.php(57): graphql_init()
#3 /var/www/html/wp-admin/includes/plugin.php(670): include_once('/var/www/html/w...')
#4 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/Plugin_Command.php(323): activate_plugin('wp-graphql-1.2....', '', NULL)
#5 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/WP_CLI/CommandWithUpgrade.php(245): Plugin_Command->activate(Array)
#6 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/Plugin_Command.php(785): WP_CLI\CommandWithUpgrade->install(Array, Array)
#7 [internal function]: Plugin_Command->install(Array, Array)
#8 phar:///usr/local/bin/wp/vendor/wp- in /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php on line 234
```

* wordpress version: 5.6.1
* other plugins: 
** wp-graphql-acf

Note that I installed the plugin via the zip file from github.

**Comment by effinrich** [comment]
> After upgrading from 1.1.3 to 1.2.5 I am getting the following error:
> 
> ```
> [02-Mar-2021 07:51:36 UTC] PHP Fatal error:  Uncaught Error: Class 'WPGraphQL\Telemetry\Tracker' not found in /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php:234
> Stack trace:
> #0 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php(76): WPGraphQL->actions()
> #1 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/wp-graphql.php(54): WPGraphQL::instance()
> #2 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/wp-graphql.php(57): graphql_init()
> #3 /var/www/html/wp-admin/includes/plugin.php(670): include_once('/var/www/html/w...')
> #4 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/Plugin_Command.php(323): activate_plugin('wp-graphql-1.2....', '', NULL)
> #5 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/WP_CLI/CommandWithUpgrade.php(245): Plugin_Command->activate(Array)
> #6 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/Plugin_Command.php(785): WP_CLI\CommandWithUpgrade->install(Array, Array)
> #7 [internal function]: Plugin_Command->install(Array, Array)
> #8 phar:///usr/local/bin/wp/vendor/wp- in /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php on line 234
> Fatal error: Uncaught Error: Class 'WPGraphQL\Telemetry\Tracker' not found in /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php:234
> Stack trace:
> #0 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php(76): WPGraphQL->actions()
> #1 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/wp-graphql.php(54): WPGraphQL::instance()
> #2 /var/www/html/wp-content/plugins/wp-graphql-1.2.5/wp-graphql.php(57): graphql_init()
> #3 /var/www/html/wp-admin/includes/plugin.php(670): include_once('/var/www/html/w...')
> #4 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/Plugin_Command.php(323): activate_plugin('wp-graphql-1.2....', '', NULL)
> #5 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/WP_CLI/CommandWithUpgrade.php(245): Plugin_Command->activate(Array)
> #6 phar:///usr/local/bin/wp/vendor/wp-cli/extension-command/src/Plugin_Command.php(785): WP_CLI\CommandWithUpgrade->install(Array, Array)
> #7 [internal function]: Plugin_Command->install(Array, Array)
> #8 phar:///usr/local/bin/wp/vendor/wp- in /var/www/html/wp-content/plugins/wp-graphql-1.2.5/src/WPGraphQL.php on line 234
> ```
> 
> * wordpress version: 5.6.1
> * other plugins:
>   ** wp-graphql-acf
> 
> Note that I installed the plugin via the zip file from github.

I just received the same error.  Wordpress 5.6.2

**Comment by jasonbahl** [comment]
@rburgst @effinrich Thanks for reporting! Checking this out. 👀

**Comment by rsm0128** [comment]
@rburgst @effinrich You should run `composer install` after cloning repository from GitHub.
@jasonbahl I created a [PR](https://github.com/wp-graphql/wp-graphql/pull/1759) to fix this issue. Please check.
Thanks!

**Comment by effinrich** [comment]
> @rburgst @effinrich You should run `composer install` after cloning repository from GitHub.
> @jasonbahl I created a [PR](https://github.com/wp-graphql/wp-graphql/pull/1759) to fix this issue. Please check.
> Thanks!

Ahhh, ok.  The tutorial I was following left that bit out.  Thanks!

**Comment by DanielLovrinov** [comment]
> @rburgst @effinrich You should run `composer install` after cloning repository from GitHub.
> @jasonbahl I created a [PR](https://github.com/wp-graphql/wp-graphql/pull/1759) to fix this issue. Please check.
> Thanks!

I don't know if that's the fix. We are downloading zip files and then uploading/installing them to wp. 

I personally followed a tutorial from the official gatsby website:
1. Download WPGraphQL and WPGraphiQL as ZIP archives from their respective GitHub repositories.
2. Upload wp-graphql-develop.zip and wp-graphiql-master.zip using the Upload Plugin button on the Add Plugins screen. You’ll have to upload and install them one at a time.

The error occurs after step 2 on activating the WP GraphQL plugin.

**Comment by rsm0128** [comment]
10 days ago (on 23th February) with commit [72dcb63](https://github.com/wp-graphql/wp-graphql/commit/72dcb63c6ec62d67d3cef170c31790262da79e7a) @jasonbahl removed the `vendor` directory and wrote a code section to add a notice in wp-admin panel.
`WPGraphQL appears to have been installed without it's dependencies. It will not work properly until dependencies are installed. This likely means you have cloned WPGraphQL from Github and need to run the command "composer install".`
And the installation guide has not been updated yet.
Also, there was a logical error with the code section, and so you guys faced such a problem. (as I mentioned above, I created a [PR](https://github.com/wp-graphql/wp-graphql/pull/1759) to fix the error)
I think @jasonbahl will update the installation guide, and fix the error.
Meanwhile, you can just run `composer install` to fix this issue.
@jasonbahl I think it's better to leave the vendor directory in repo. Without that all users **have to** install `composer` to use this **plugin**.
I know there are some guys who says having vendor directory in repo is unprofessional, but this **WPGraphQL** is a **WordPress plugin** and users just want it work as soon as they **download** and **install**.

**Comment by rburgst** [comment]
> @jasonbahl I think it's better to leave the vendor directory in repo. Without that all users **have to** install `composer` to use this **plugin**.
> I know there are some guys who says having vendor directory in repo is unprofessional, but this **WPGraphQL** is a **WordPress plugin** and users just want it work as soon as they **download** and **install**.

I totally agree. This is the way it works for *all the other wordpress plugins* as well.

**Comment by jasonbahl** [comment]
For quite a while the plan (see: #933) has been to follow Composer best practices and not version the vendor dependencies once the plugin was released on WordPress.org. 

Vendor dependencies were versioned so that the plugin could be downloaded as a zip from Github and be used without composer, but now that the plugin is available on WordPress.org, that's the primary distribution channel. 

Currently, Github should be used for developers that want to contribute to the plugin, and users that want to install the plugin should get it from one of the following recommended distribution channels: 

- WordPress.org: https://wordpress.org/plugins/wp-graphql/
- Packagist.org: https://packagist.org/packages/wp-graphql/wp-graphql
- WPackagist.org: https://wordpress.org/plugins/wp-graphql/

I've updated the Quick Start guide to reflect this.
